src/DotNetty.Handlers/Tls/ServerTlsSniSettings.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace DotNetty.Handlers.Tls
+{
+    using System;
+    using System.Diagnostics.Contracts;
+    using System.Security.Authentication;
+    using System.Security.Cryptography.X509Certificates;
+
+    public sealed class ServerTlsSniSettings : TlsSettings
+    {
+        public ServerTlsSniSettings(Func<string, X509Certificate2> serverCertificateSelector)
```

Comment:
>Func<string, X509Certificate2> [](start = 36, length = 30)

pls change to Func<string, Task<X509Certificate2>>. I can imagine pulling cert out of remote store / from disk lazily.